packages/@aws-cdk/aws-appmesh/lib/service-discovery.ts

```diff
@@ -0,0 +1,110 @@
+import * as cloudmap from '@aws-cdk/aws-servicediscovery';
+import * as cdk from '@aws-cdk/core';
+import { CfnVirtualNode } from './appmesh.generated';
+
+
+/**
+ * Represents the properties needed to define CloudMap Service Discovery
+ */
+export interface CloudMapServiceDiscoveryOptions {
+  /**
+   * The AWS Cloud Map Service to use for service discovery
+   */
+  readonly service: cloudmap.IService;
+
+  /**
+   * A string map that contains attributes with values that you can use to
+   * filter instances by any custom attribute that you specified when you
+   * registered the instance. Only instances that match all of the specified
+   * key/value pairs will be returned.
+   *
+   * @default - no instance attributes
+   */
+  readonly instanceAttributes?: {[key: string]: string};
+}
+
+/**
+ * Properties for VirtualNode Service Discovery
+ */
+export interface ServiceDiscoveryConfig {
+  /**
+   * DNS based Service Discovery
+   *
+   * @default - no DNS based service discovery
+   */
+  readonly dns?: CfnVirtualNode.DnsServiceDiscoveryProperty;
+
+  /**
+   * Cloud Map based Service Discovery
+   *
+   * @default - no Cloud Map based service discovery
+   */
+  readonly cloudmap?: CfnVirtualNode.AwsCloudMapServiceDiscoveryProperty;
+}
+
+/**
+ * Provides the Service Discovery method a VirtualNode uses
+ */
+export abstract class ServiceDiscovery {
+  /**
+   * Returns DNS based service discovery
+   */
+  public static dns(hostname: string): ServiceDiscovery {
+    return new DnsServiceDiscovery(hostname);
+  }
+
+  /**
+   * Returns Cloud Map based service discovery
+   */
+  public static cloudmap(options: CloudMapServiceDiscoveryOptions): ServiceDiscovery {
```

Comment:
Should this be called `cloudMap` instead? That seems to be the casing we usually use for this service, and it's also called "Cloud Map", not "Cloudmap".

Reply:
yep! Good catch. I'll make that change